src/main/kotlin/com/autonomousapps/tasks/ReasonTask.kt

```diff
@@ -216,7 +216,8 @@ abstract class ReasonTask @Inject constructor(
         }?.gav()
 
       // Guaranteed to find full GAV or throw
-      val gavKey = dependencyUsages.entries.find(requestedId::matchesKey)?.key
+      val filteredDependencyKey = findFilteredDependencyKey(requestedId)
+      val gavKey = filteredDependencyKey
         ?: annotationProcessorUsages.entries.find(requestedId::matchesKey)?.key
```

Comment:
@autonomousapps it seems that this lenient filter can have the same problem. WDYT we can use the same logic here? Or let's keep it out of scope?

Reply:
No, let's fix both.

Reply:
done